internal/pkg/platform/platform_matcher.go

```diff
@@ -82,9 +83,25 @@ func getCPUVariantWindows(arch string) string {
 func getCPUVariantArm() string {
 	variant, err := getCPUInfo("Cpu architecture")
 	if err != nil {
+		logrus.Errorf("Couldn't get cpu architecture")
 		return ""
 	}
-	// TODO handle RPi Zero mismatch (https://github.com/moby/moby/pull/36121#issuecomment-398328286)
+
+	// handle RPi Zero variant mismatch due to wrong variant from kernel
+	// https://github.com/containerd/containerd/pull/4530
+	// https://www.raspberrypi.org/forums/viewtopic.php?t=12614
+	// https://github.com/moby/moby/pull/36121#issuecomment-398328286
+	if runtime.GOARCH == "arm" && variant == "7" {
+		model, err := getCPUInfo("model name")
```

Comment:
Non-blocking:

Plausibly `getCPUInfo` could parse everything at once and return a `map[key]value`. That would remove some of the error paths….